forge-game/src/main/java/forge/game/card/CardProperty.java

```diff
@@ -1529,6 +1526,9 @@ else if (prop.isEmpty() && dmgSource.equalsWithGameTimestamp(source)) {
             } else if (property.startsWith("numColors")) {
                 rhs = property.substring(11);
                 y = card.getColor().countColors();
+            } else if (property.startsWith("numTypes")) {
+                rhs = property.substring(10);
+                y = ((Collection<?>)card.getType().getCoreTypes()).size();
```

Comment:
this is bad for type safety
use `Iterables.size`

Reply:
Big mistake, thanks for the catch!